android-sdk/src/main/java/co/elastic/apm/android/sdk/traces/otel/processor/ElasticSpanProcessor.java

```diff
@@ -30,23 +32,27 @@
 public class ElasticSpanProcessor implements SpanProcessor {
     private final SpanProcessor original;
     private final Set<ExclusionRule> rules = new HashSet<>();
+    private final SessionIdProvider sessionIdProvider;
+    private static final String SESSION_ID_ATTRIBUTE_KEY = "session.id";
 
     public void addAllExclusionRules(Collection<? extends ExclusionRule> rules) {
         this.rules.addAll(rules);
     }
 
     public ElasticSpanProcessor(SpanProcessor original) {
         this.original = original;
+        sessionIdProvider = ElasticApmAgent.get().configuration.sessionIdProvider;
     }
 
     @Override
     public void onStart(Context parentContext, ReadWriteSpan span) {
+        span.setAttribute(SESSION_ID_ATTRIBUTE_KEY, sessionIdProvider.getSessionId());
```

Comment:
That's a good point, I just checked and it seems like it's not thread safe according to this: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk.md#onstart - So I'll update it in here to avoid thread issues

Reply:
I've just pushed some updates for it